src/vs/workbench/contrib/positronNotebook/browser/PositronNotebookInstance.ts

```diff
@@ -975,6 +992,144 @@ export class PositronNotebookInstance extends Disposable implements IPositronNot
 		}
 	}
 
+
+	// #endregion
+
+	// =============================================================================================
+	// #region Clipboard Methods
+
+	/**
+	 * Internal clipboard for storing cells with full fidelity
+	 */
+	private _clipboardCells: ICellDto2[] = [];
+
+	/**
+	 * Flag to track if the clipboard contains cut cells (vs copied cells)
+	 */
+	private _isClipboardCut: boolean = false;
+	copyCells(cells?: IPositronNotebookCell[]): void {
+		const cellsToCopy = cells || this.getSelectedCells();
+
+		if (cellsToCopy.length === 0) {
+			return;
+		}
+
+		// Store internally for full-fidelity paste
+		this._clipboardCells = cellsToCopy.map(cell => cellToCellDto2(cell));
+		this._isClipboardCut = false;
+
+		// Also write to system clipboard as text
+		const clipboardText = serializeCellsToClipboard(cellsToCopy);
```

Comment:
This is interesting! For future: We could use some other serialization format that might be more helpful for data scientists than the raw JSON and/or make that configurable. Maybe some Markdown formatted version of the cell source and outputs, or something like what's in the console eg:

```
>>> x = 1
>>> x + 1
2
```